great_expectations/datasource/fluent/pandas_azure_blob_storage_datasource.py

```diff
@@ -55,11 +59,19 @@ class PandasAzureBlobStorageDatasource(_PandasFilePathDatasource):
     def _get_azure_client(self) -> azure.BlobServiceClient:
         azure_client: Union[azure.BlobServiceClient, None] = self._azure_client
         if not azure_client:
+            _check_config_substitutions_needed(
+                self, self.azure_options, raise_warning_if_provider_not_present=True
+            )
+            # pull in needed config substitutions using the `_config_provider`
+            azure_options: dict = self.dict(config_provider=self._config_provider)[
```

Comment:
@Kilo59 Please make a small docstring/comment that the Pydantic `dict()` in line 66 is what performs the substitutions and hence absorbs `self.azure_options`.  Thank you!